nixos/modules/services/web-apps/pixelfed.nix

```diff
@@ -391,6 +392,9 @@ in {
         rsync -av --no-perms ${pixelfed}/storage-static/ ${cfg.dataDir}/storage
         chmod -R +w ${cfg.dataDir}/storage
 
+        chmod g+x ${cfg.dataDir}/storage ${cfg.dataDir}/storage/app
+        chmod -R g+rX ${cfg.dataDir}/storage/app/public
```

Comment:
This can take a long time to perform… :/

Reply:
Indeed, I guess only doing it once and recording it via the presence of a `.public-data` file would be better

Reply:
systemd-tmpfiles could ensure those permissions ideally.

Reply:
Wouldn't it take the same time as doing a chmod operation ? If it has to check every file it seems to have no gain

Reply:
The most important thing here is correctness anyway and it seems more fragile to do the chmod/chown dance ourselves to ensure they are there rather than letting systemd doing the work, which can be as slow as doing it ourselves but maybe faster?

Reply:
Regarding using systemd.tmpfiles for file permissions, I thing that it is run before starting the different services, in that case the files won't be present yet to have their permissions changed

Reply:
Furthermore, when using `Z` to apply permissions recursively there is no difference between files and directories, in contrast to `chmod +X`, which would be a problem

Reply:
You convinced me ;)